File: src/api/FunCoding.ReVision/FunCoding.ReVision.WebApi/Models/ComposeRequest.cs
namespace FunCoding.ReVision.WebApi.Models;

public class ComposeRequest
{
    public string CurrentEmailSubject { get; set; } = string.Empty;
    public string CurrentEmailBody { get; set; } = string.Empty;
    public string Input { get; set; } = string.Empty;
    public string TargetLanguage { get; set; } = string.Empty;
    public string WritingTone { get; set; } = string.Empty;
}

File: src/outlook-add-in/ReVision/src/taskpane/models/composeRequest.ts
export class ComposeRequest {
  currentEmailSubject: string;
  currentEmailBody: string;
  input: string;
  targetLanguage: string;
  writingTone: string;
  constructor(
    currentEmailSubject: string,
    currentEmailBody: string,
    input: string,
    targetLanguage: string,
    writingTone: string
  ) {
    this.currentEmailSubject = currentEmailSubject;
    this.currentEmailBody = currentEmailBody;
    this.input = input;
    this.targetLanguage = targetLanguage;
    this.writingTone = writingTone;
  }
}
